sycl/include/sycl/group.hpp

```diff
@@ -301,7 +301,7 @@ template <int Dimensions = 1> class __SYCL_TYPE(group) group {
   /// Permitted types for dataT are all scalar and vector types, except boolean.
   template <typename dataT>
   std::enable_if_t<!detail::is_bool<dataT>::value, device_event>
-  async_work_group_copy(local_ptr<dataT> dest, global_ptr<dataT> src,
+  async_work_group_copy(local_ptr<dataT> dest, global_ptr<const dataT> src,
```

Comment:
While playing around with these, I noticed that this cannot trigger the implicit conversion from non-const to const as that is not how template argument deduction works.

Alternatively, what we could do here is change it so that `src` is either `dataT` or `const dataT`, and then do the proper conversion inside here. That is, something like:

```c++
  template <typename destDataT, typename srcDataT>
  std::enable_if_t<!detail::is_bool<destDataT>::value &&
                   std::is_same_v<destDataT, std::remove_const_t<srcDataT>>, device_event>
  async_work_group_copy(local_ptr<destDataT> dest, global_ptr<srcDataT> src,
```

and then change the `SrcT` alias below to something like

```c++
using SrcT = detail::ConvertToOpenCLType_t<global_ptr<const srcDataT>>;
```

Reply:
Since these interfaces are old and not part of SYCL 2020, I have opened a separate PR with the SYCL 2020 equivalents (deprecating the old ones): https://github.com/intel/llvm/pull/9377. Once that is in, I don't think we need to change the `async_work_group_copy` interfaces in this PR.